File: problems/204. Count Primes/solution.py
class Solution:

    def countPrimes(self, n: int):
        table = [0]*n
        count = 0

        for num in range(2, n):
            if table[num] == 1:
                continue
            count += 1
            temp = num*num
            while temp < n:
                table[temp] = 1
                temp += num
        return count

File: problems/204. Count Primes/solution.cpp
class Solution
{
public:
    int countPrimes(int n)
    {
        int count = 0;
        vector<int> visited(n);
        for (int i = 0; i < n; i++)
            visited[i] = false;

        for (unsigned long i = 2; i < n; i++)
        {
            if (visited[i] == true)
                continue;
            count++;
            unsigned long temp = i * i;
            while (temp < n)
            {
                visited[temp] = true;
                temp += i;
            }
        }

        return count;
    }
};